module/zfs/zvol.c

```diff
@@ -65,7 +65,17 @@ static list_t zvol_state_list;
 #define	ZVOL_HT_SIZE	1024
 static struct hlist_head *zvol_htable;
 #define	ZVOL_HT_HEAD(hash)	(&zvol_htable[(hash) & (ZVOL_HT_SIZE-1)])
-static DEFINE_IDA(zvol_ida);
+
+/*
+ * Note: this is an unrolled version of the DEFINE_IDA() macro,
+ * modified to not use a cast.  The cast does not compile with the
+ * compiler flags used for ZoL (-std=gnu99).
```

Comment:
With `-std=gnu99`, with the cast, it gets that error, and without the cast it does not.

Without `-std=gnu99`, it works (with the cast, i.e. there is no bug in MASTER).